extensions/wikia/Recirculation/js/helpers/recommendedContent.js

```diff
@@ -0,0 +1,107 @@
+/*global define*/
+define('ext.wikia.recirculation.helpers.recommendedContent', [
+    'jquery',
+    'wikia.window',
+    'wikia.nirvana',
+    'ext.wikia.recirculation.helpers.blacklist',
+    'wikia.log',
+    'wikia.eventLogger'
+], function ($, w, nirvana, blacklist, log, eventLogger) {
+    'use strict';
+
+    var numberOfArticleFooterSlots = $('#mixed-content-footer').data('number-of-wiki-articles');
+    var REQUEST_ID;
+
+    function getFilteredItems(response) {
+        var blacklistedItems = blacklist.get();
+
+        return response.article_recommendation
+            .filter(function (el) {
+                return blacklistedItems.indexOf(el.item_id) === -1;
+            })
+            .concat(response.wiki_recommendation);
+    }
+
+    function mapExperimentalDataResponse(response) {
+        var filteredItems = getFilteredItems(response);
+
+        REQUEST_ID = response.recommendation_request_id;
+
+        if (filteredItems < numberOfArticleFooterSlots) {
+            blacklist.remove(5);
```

Comment:
Why 5?

Reply:
I've assigned it to a variable, 5 seemed reasonable for me as we'd have slightly more recommendations to choose from then